alphagov/paas-cf
# [#163264158]Set firehose.doppler.idle_timeout to enable re-establishing firehose connection

What
----

Sets the [`doppler.idle_timeout`](https://github.com/bosh-prometheus/firehose_exporter#flags) setting so that firehose websocket connections are re-established if they're idle for too long.

How to review
-------------
To replicate without the fix
```
cd paas-bootstrap
aws-paas-dev make dev bosh-cli
export BOSH_DEPLOYMENT=prometheus
bosh vms
bosh ssh <firehose vm>
# Try to simulate a network outage by blocking incoming
# and outgoing traffic to the doppler hostname on the firehouse vm
# Wait at least 6 minutes for the buffer to dry up
sudo iptables -A INPUT -s doppler.<env name>.dev.cloudpipeline.digital -j DROP
sudo iptables -A OUTPUT -s doppler.<env name>.dev.cloudpipeline.digital -j DROP

# Once the buffer has dried up, unblock the connection
# The ELB has two instances, so two iptables rules get added
sudo iptables -D INPUT 1
sudo iptables -D INPUT 1
sudo iptables -D OUTPUT 1
sudo iptables -D OUTPUT 1

# Should see that metrics don’t go back up
```

To verify with the fix
1. Repeat replication steps up to the ip tables rules having been added
2. Wait about 15 minutes (5 minutes for the buffer, 5 minutes for idle timeout, 5 minutes for good measure)
3. Remove the ip tables rules
4. See that metrics are returned shortly afterwards

Who can review
--------------
Not @AP-Hunt
